pyro/nn/module.py

```diff
@@ -77,7 +78,40 @@ def _get_pyro_params(module):
         yield name, module._parameters[name]
 
 
-class PyroModule(torch.nn.Module):
+# Inherit from ABCMeta because many practical nn.Module subclasses also inherit
+# from ABCMeta (e.g. EasyGuide), and this will avoid metclass conflict errors.
+class _PyroModuleMeta(ABCMeta):
+    _pyro_mixin_cache = {}
+
+    # Unpickling helper to create an empty object of type PyroModule[Module].
+    class _New:
```

Comment:
I don't fully understand all the magic that happens here, but I have done a few tests and have been very satisfied with what I got. :)